static/app/views/projectInstall/issueAlertNotificationOptions.tsx

```diff
@@ -0,0 +1,277 @@
+              id: IssueAlertActionType.SLACK,
+              workspace: integration?.id,
+              channel: channel,
+            },
+          ];
+          break;
+        case 'discord':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.DISCORD,
+              server: integration?.id,
+              channel_id: channel,
+            },
+          ];
+          break;
+        case 'msteams':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.MS_TEAMS,
+              team: integration?.id,
+              channel: channel,
+            },
+          ];
+          break;
+        default:
+          return undefined;
+      }
+
+      return api.requestPromise(`/projects/${organization.slug}/${projectSlug}/rules/`, {
+        method: 'POST',
+        data: {
+          name,
+          conditions,
+          actions: integrationAction,
+          actionMatch,
+          frequency,
+        },
+      });
+    },
+    [
+      actions,
+      api,
+      provider,
+      integration,
+      channel,
+      organization.features,
+      organization.slug,
+    ]
+  );
+
+  return {
+    createNotificationAction,
+    notificationProps: {
+      actions,
+      provider,
+      integration,
+      channel,
+      setActions,
+      setProvider,
+      setIntegration,
+      setChannel,
+    },
+  };
+}
+
+export default function IssueAlertNotificationOptions(
+  notificationProps: IssueAlertNotificationProps
+) {
+  const organization = useOrganization();
+  const {actions, provider, setActions, setIntegration, setProvider} = notificationProps;
+
+  const messagingIntegrationsQuery = useApiQuery<OrganizationIntegration[]>(
+    [`/organizations/${organization.slug}/integrations/?integrationType=messaging`],
+    {staleTime: Infinity}
+  );
+
+  const providersToIntegrations = useMemo(() => {
+    const map: {[key: string]: OrganizationIntegration[]} = {};
+    if (messagingIntegrationsQuery.data) {
+      for (const i of messagingIntegrationsQuery.data) {
+        const providerSlug = i.provider.slug;
+        map[providerSlug] = map[providerSlug] ?? [];
+        map[providerSlug].push(i);
+      }
+    }
+    return map;
+  }, [messagingIntegrationsQuery.data]);
+
+  useEffect(() => {
+    const providerKeys = Object.keys(providersToIntegrations);
+    if (providerKeys.length > 0) {
+      const firstProvider = providerKeys[0];
+      setProvider(firstProvider);
+
+      const firstIntegration = providersToIntegrations[firstProvider][0];
+      setIntegration(firstIntegration);
+    }
+  }, [providersToIntegrations, setProvider, setIntegration]);
+
+  const shouldRenderSetupButton = useMemo(() => {
+    return messagingIntegrationsQuery.data?.every(i => i.status !== 'active');
+  }, [messagingIntegrationsQuery.data]);
+
+  const shouldRenderNotificationConfigs = useMemo(() => {
+    return actions.some(v => v !== MultipleCheckboxOptions.EMAIL);
+  }, [actions]);
```

Comment:
Same here, no need for `useMemo`.

use memo should really only be used when we 

1. need a stable reference to a value where it would change if it was re-computed on every render: eg if we had

   ```tsx
   const something = actions.map(...)
   ```

   This would create a new object every time, and we would want to put it in a memo to stop that from happening

OR

2. The computation is expensive. This is actually quite rare.